webapp/src/main/java/com/box/l10n/mojito/service/tm/PluralNameParser.java

```diff
@@ -2,10 +2,28 @@
 
 import org.springframework.stereotype.Component;
 
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 @Component
 public class PluralNameParser {
 
     public String getPrefix(String nameWithPluralForm) {
         return nameWithPluralForm.replaceAll("_(zero|one|two|few|many|other)$", "");
     }
+
+    public static String getPrefix(String name, String separator) {
+
+        Pattern pattern = Pattern.compile("(.*)" + separator + "(zero|one|two|few|many|other)");
```

Comment:
why static inside the @Component ? code could be shared?

Reply:
It doesn't use any instance variable, so no need to use an instance method. I've added these methods here since its already doing something along the same lines.

Reply:
I guess the point to have a spring component is for potential testing. If this is irrelevant/overkill, the @Component could be removed and all methods made static. But mixing both style don't look great.

Reply:
Updated.